### ISSUE TITLE
Implement floating disks attach and detach action

### ISSUE THREAD

**Issue by pdilung** [abenari/rbovirt#60]
Implement attach and detach action for _floating_ disks according to RH [Technical Guide](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Virtualization/3.4/html/Technical_Guide/sect-Sub-Collections6.html) for RHEV 3.4.

This will likely require to implement:
* __disk_api__ - an API for creation of `OVIRT::Volume` instances on per entry-point and/or datacenter  level.
* Following methods in __vm_api__:
    * `OVIRT::Client#volume_active?`
    * `OVIRT::Client#activate_volume`
    * `OVIRT::Client#deactivate_volume`
    * `OVIRT::Client#attach_volume`
    * `OVIRT::Client#detach_volume`
* Implement new and/or enrich other methods that are required by methods mentioned in the item above.

**Comment by pdilung** [comment]
I done an initial implementation. It can be found at 861917b87268d595edd5f33277d9974e81652d65 @abenari: Thanks a lot for the __disks_api__ code.

**Comment by pdilung** [comment]
Hello All,

The disk detach action API call, decsibed in [RHEV Technical Guide](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Virtualization/3.4/html/Technical_Guide/sect-Sub-Collections6.html#sect-Disks_Sub-Collection) and shown in the listing bellowow, uses *DELETE method with a message body*.

```xml
DELETE /api/vms/082c794b-771f-452f-83c9-b2b5a19c0399/disks/
  d135f1c5-b5e1-4238-9381-b3277f5a3742HTTP/1.1
Accept: application/xml
Content-type: application/xml

<action>
    <detach>true</detach>
</action>
```

In order to implement disk detach action, I had to re-implement `OVIRT::Client#http_delete` as well as `RestClient::Resource#delete` method.

The problem is that this has required *monkey patching* of a method that belongs to `rest-client` package. This package is very likely widely used and introducing a _payload_ in `delete` method would have caused incompatibilities in many existing applications and thus I see very little chance to get this implemented in `rest-client`'s upstream. It would effectively mean that the `delete` method would have to be maintained within `rbovirt`.

Initially, I wanted to write a bug report to RH and/or to oVirt that a disk detachment should be done by DELETE request with URI that would contain a query string:
```
DELETE /api/vms/082c794b-771f-452f-83c9-b2b5a19c0399/disks/
  d135f1c5-b5e1-4238-9381-b3277f5a3742?detach=true
```
However after I have skimmed through [RFC 2616](http://www.w3.org/Protocols/rfc2616/rfc2616.html) and [this post](http://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request) it seems to me that the current specification does not explicitly and strictly allows nor forbids bodies on DELETE requests. There is draft [HTTP/1.1, part 2: Message Semantics](http://tools.ietf.org/html/draft-ietf-httpbis-p2-semantics-19#page-23) that, if approved, explicitly states that:
> Bodies on DELETE requests have no defined semantics.  Note that sending a body on a DELETE request might cause some existing implementations to reject the request.

Given that the RFC is not really 100% clear on DELETE I am not sure whether RHEV/oVirt guys will be looking forward to change anything in the current implementation of disk detach action API call.

For the time being, I would suggest the following:
1. Keep `RestClient::Resource#delete` monkey patched.
2. Modify the `OVIRT::Client#http_delete` to explicitly support payload so it will have same parameters as methods for GET, POST and PUT and adopt all (currently 4) callouts to this method. My current implementation (see 861917b87268d595edd5f33277d9974e81652d65) tries to mimic the old behavior yet supports body in requests. But I can rewrite this so all HTTP methods will have same parameters.

@abenari: Would be great if you shared your opinion and suggestions. If you agree I can create a merge request.

**Comment by abenari** [comment]
LGTM,
Send a pull request, that will let everyone that watch this repository a chance to comment.

**Comment by pdilung** [comment]
Pull request created as #61.

**Comment by abenari** [comment]
Fixed in 480edf68a69c84bf5b85c4ae1130ef8402662a8a thanks @pdilung
